integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/rule/InvalidRowTotalCountRuleImpl.java

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thinkbiganalytics.spark.dataquality.util.DataQualityConstants;
+import com.thinkbiganalytics.spark.dataquality.util.FlowAttributes;
+import com.thinkbiganalytics.spark.dataquality.util.MissingAttributeException;
+
+import org.json.simple.JSONObject;
+
+/**
+ * Data Quality Rule to ensure that: <br>
+ * Number of Invalid rows <= invalid threshold defined by the INVALID_THRESHOLD_ATTRIBUTE
+ * 
+ * If no threshold attribute is defined, the default value is used which is 0 i.e. no invalid roes
+ */
+public class InvalidRowTotalCountRuleImpl implements DataQualityRule {
+    private static final Logger log = LoggerFactory.getLogger(InvalidRowTotalCountRuleImpl.class);
+
+    private String name;
+    private String description;
+    private boolean status;
+
+    long invalidRowCount;
+    long invalidThreshold;
+
+    public InvalidRowTotalCountRuleImpl() {
+        this.name = "INVALID_ROW_TOTAL_COUNT_RULE";
+        this.description = "Number of invalid rows less than set threshold value in " +
+                           DataQualityConstants.DQ_INVALID_ALLOWED_COUNT_ATTRIBUTE;
+        this.status = false;
+    }
+
+    @Override
+    public boolean loadAttributes(FlowAttributes flowAttr) {
+        try {
+            invalidRowCount = flowAttr.getAttributeValueLong(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE);
+            invalidThreshold = flowAttr.getAttributeValueLong(DataQualityConstants.DQ_INVALID_ALLOWED_COUNT_ATTRIBUTE,
+                                                              DataQualityConstants.DEFAULT_INVALID_ALLOWED_COUNT_VALUE);
+        } catch (MissingAttributeException e) {
+            log.error("Required attributes missing");
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public boolean getStatus() {
+        return status;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public JSONObject getSummary() {
+
+        JSONObject jsonAttrs = new JSONObject();
+        jsonAttrs.put("Invalid Row Count", invalidRowCount);
+        jsonAttrs.put("Invalid Threshold", invalidThreshold);
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("DESCRIPTION", getDescription());
+        jsonObject.put("STATUS", status);
+        jsonObject.put("ATTRIBUTES", jsonAttrs);
+
+        return jsonObject;
+    }
+
+    @Override
+    public boolean evaluate() {
+        try {
+            status = (invalidRowCount <= invalidThreshold);
+
+            if (!status) {
+                log.error("Invalid Row Count = " + invalidRowCount
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)